spec/github-pages-configuration_spec.rb

```diff
@@ -1,10 +1,18 @@
 require "spec_helper"
 
 describe(GitHubPages::Configuration) do
-  let(:test_config)   { { "source" => fixture_dir, "quiet" => true, "testing" => "123" } }
+  let(:test_config) do
+    {
+      "source" => fixture_dir,
+      "quiet" => true,
+      "testing" => "123",
+      "destination" => tmp_dir }
+  end
   let(:configuration) { Jekyll.configuration(test_config) }
   let(:site)          { Jekyll::Site.new(configuration) }
   let(:effective_config) { described_class.effective_config(site.config) }
+  before { ENV.delete("DISABLE_WHITELIST") }
+  before { ENV["JEKYLL_ENV"] = "test" }
```

Comment:
`before` is generally seen as bad to use, because it's not specific. `before(:all)` or `before(:each)` would be preferable.